darts/models/forecasting/regression_ensemble_model.py

```diff
@@ -148,10 +219,44 @@ def fit(
             model.fit(**kwargs)
         return self
 
+    def predict(
+        self,
+        n: int,
+        series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        num_samples: int = 1,
+        verbose: bool = False,
+    ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+
+        super().predict(
+            n=n,
+            series=series,
+            past_covariates=past_covariates,
+            future_covariates=future_covariates,
+            num_samples=num_samples,
+            verbose=verbose,
+        )
+
+        # support deterministic forecasting model with probabilistic ensembling
+        predictions = self._make_multiple_predictions(
+            n=n,
+            series=series,
+            past_covariates=past_covariates,
+            future_covariates=future_covariates,
+            num_samples=self.regression_train_num_samples,
+        )
+
+        if predictions[0].n_samples > 1:
```

Comment:
I believe we can move `predict()` and `_predictions_reduction()` back to the base `EnsembleModel`.
- For the naive ensemble, we can set the `regression_train_samples_reduction=None` in the `__init__()` body. Then when calling super().predict() we set num_samples=1.

Reply:
Moved `predict()` and  `_predictions_reduction()` back to `EnsembleModel`, which now accept two new arguments in it constructor (`train_num_samples` and `train_samples_reduction`). They might be reusable during implemention of new ensemble training techniques such as boosting but are ignored in "single level" models such as `NaiveEnsemble`.

`predict()` relies on the value of the `reduction_method` attribute to know if it should use `num_samples` directly (to have a probabilistic forecast with `NaiveEnsemble`) or `regression_train_n_samples` so that the `future_covariates` passed to the `RegressionEnsembleModel` regressor have the dimensions seen during the training.